crates/astria-sequencer/src/bridge/query.rs

```diff
@@ -15,6 +18,153 @@ use crate::{
+    if err.is_none() {
+        return response::Query {
+            code: code.into(),
+            info: code.to_string(),
+            log: info.into(),
+            ..response::Query::default()
+        };
+    }
+
+    let err = err.unwrap();
+    response::Query {
+        code: code.into(),
+        info: code.to_string(),
+        log: format!("{info}: {err:#}"),
+        ..response::Query::default()
+    }
+}
+
+async fn get_bridge_account_info(
+    snapshot: cnidarium::Snapshot,
+    address: Address,
+) -> anyhow::Result<Option<BridgeAccountInfo>, response::Query> {
+    let rollup_id = match snapshot.get_bridge_account_rollup_id(&address).await {
+        Ok(Some(rollup_id)) => rollup_id,
+        Ok(None) => {
+            return Ok(None);
+        }
+        Err(err) => {
+            return Err(error_query_response(
+                Some(err),
+                AbciErrorCode::INTERNAL_ERROR,
+                "failed to get rollup id",
+            ));
+        }
+    };
+
+    let asset_id = match snapshot.get_bridge_account_asset_id(&address).await {
+        Ok(asset_id) => asset_id,
+        Err(err) => {
+            return Err(error_query_response(
+                Some(err),
+                AbciErrorCode::INTERNAL_ERROR,
+                "failed to get asset id",
+            ));
+        }
+    };
+
+    let sudo_address = match snapshot.get_bridge_account_sudo_address(&address).await {
+        Ok(Some(sudo_address)) => sudo_address,
+        Ok(None) => {
+            return Err(error_query_response(
+                None,
+                AbciErrorCode::INTERNAL_ERROR,
+                "sudo address not set",
+            ));
+        }
+        Err(err) => {
+            return Err(error_query_response(
+                Some(err),
+                AbciErrorCode::INTERNAL_ERROR,
+                "failed to get sudo address",
+            ));
+        }
+    };
+
+    let withdrawer_address = match snapshot
+        .get_bridge_account_withdrawer_address(&address)
+        .await
+    {
+        Ok(Some(withdrawer_address)) => withdrawer_address,
+        Ok(None) => {
+            return Err(error_query_response(
+                None,
+                AbciErrorCode::INTERNAL_ERROR,
+                "withdrawer address not set",
+            ));
+        }
+        Err(err) => {
+            return Err(error_query_response(
+                Some(err),
+                AbciErrorCode::INTERNAL_ERROR,
+                "failed to get withdrawer address",
+            ));
+        }
+    };
+
+    Ok(Some(BridgeAccountInfo {
+        rollup_id,
+        asset_id,
+        sudo_address,
+        withdrawer_address,
+    }))
+}
+
+pub(crate) async fn bridge_account_info_request(
+    storage: Storage,
+    request: request::Query,
+    params: Vec<(String, String)>,
+) -> response::Query {
+    use astria_core::protocol::bridge::v1alpha1::BridgeAccountInfoResponse;
+
+    let address = match preprocess_request(&params) {
```

Comment:
I think you can apply the `?` operator here because this method returns a `response::Query`?

Also, other queries check the address prefix against the globally configured address prefixes, see the `astria_sequencer::address` submodule.

Should this also do that?

Reply:
> I think you can apply the ? operator here because this method returns a response::Query?

doesn't work since this function doesn't return a result, just a `Query` :/

> Should this also do that?

yes, will add

Reply:
> doesn't work since this function doesn't return a result, just a Query :/

Ah, yeah. :S